Program.cs

```diff
@@ -1,7 +1,144 @@
+      /// <param name="item"></param>
+      /// <returns>Returns the boolean value of presence and absence of the particular element in the list</returns>
+      public bool Remove (T item) {
+         int index = Array.IndexOf (myList, item);
+         if (index != -1)
+            RemoveAt (index);
+         return false;
+      }
+
+      /// <summary>Method used to clear all the elements in the list</summary>
+      public void Clear () {
+         Array.Clear (myList);
+         mCount = 0;
+      }
+
+      /// <summary>Method insert the given element at the particular index given</summary>
+      /// <param name="index">Index in which element should be inserted</param>
+      /// <param name="item">Element to be inserted</param>
+      /// <exception cref="ArgumentOutOfRangeException">If the index passed is out of range, it throws argument out of range exception</exception>
+      public void Insert (int index, T item) {
+         if (CheckValidArgumentIndex (index, mCount)) {
+            if (mCount == mCapacity)
+               ResizeArray ();
+            for (int i = mCount; i > index; i--)
+               myList[i] = myList[i - 1];
+            myList[index] = item;
+            mCount++;
+         }
+      }
+
+      /// <summary>Method remove the element in the particular index given</summary>
+      /// <param name="index">Index in which elemnt to be removed</param>
+      public void RemoveAt (int index) {
+         if (CheckValidArgumentIndex (index, mCount - 1)) {
+            for (int i = index; i < mCount - 1; i++)
+               myList[i] = myList[i + 1];
+            mCount--;
+         }
+      }
+
+      /// <summary>Method will double the size of list if the count exceeds the capacity</summary>
+      private void ResizeArray () {
+         mCapacity *= 2;
+         Array.Resize (ref myList, mCapacity);
+      }
+
+      /// <summary>This method check whether given index is within the limit of list or exceeds it</summary>
+      /// <param name="index">Index to be checked</param>
+      /// <returns>Returns true if index is within limit, false otherwise</returns>
+      /// <exception cref="IndexOutOfRangeException">If index is out of limit, throw index out of range exception</exception>
+      public void CheckValidIndex (int index) {
+         if (index < 0 || index >= mCount)
+            throw new IndexOutOfRangeException ("Index out of range");
+      }
+
+      /// <summary>Method check whether argument to be passed at partuclar index is within the limit of list</summary>
+      /// <param name="index">Index of the argument to be passed</param>
+      /// <returns>Returns true if index of element is within the limit or false otherwise</returns>
+      /// <exception cref="ArgumentOutOfRangeException">If index is out of limit, throw argument out of range exception</exception>
+      public bool CheckValidArgumentIndex (int index, int mCount) {

```

Comment:
```suggestion
      public bool CheckValidArgumentIndex (int index) {
```

mCount can be accessed in this method, without passing it as an argument